lib/identity_job_log_subscriber.rb

```diff
@@ -179,6 +218,15 @@ def trace_id(job)
   def duplicate_cron_error?(ex)
     ex.is_a?(ActiveRecord::RecordNotUnique) && ex.message.include?('(cron_key, cron_at)')
   end
+
+  def should_error?(job, ex)
+    return true if ex.nil?
+
+    !(job.class.try(:warning_messages)&.include(ex.class.name) ||
+      RiscDeliveryJob.warning_error_classes(job).any? { |warning_class|
+        ex.class.is_a? warning_class
+      })
+  end
```

Comment:
1. These two checks do the same thing, we should not need to hardcode `RiscDeliveryJob` here given that we have an interface now
2. Also switching to `none?` so we don't need to negate the `.any?`
    - Another idea would be to switch this method to `should_warn?` and leave it as an `any?`
3. `.is_a?` is called from an instance to a class, so we want `ex.is_a?(warning_class)` here

```suggestion
      job.class.warning_error_classes.none? do |warning_class|
        ex.is_a?(warning_class)
      end
  end
```